### PR TITLE
feat: implementing minimal cache wrapper

### PR DESCRIPTION
**Pull request by samgj18**

### CODE REVIEW

File: integrationos-api/src/middleware/auth.rs [hunk 1]
```diff
@@ -21,48 +24,41 @@ pub async fn auth(
         // auth header value starts with either id_ or sk_ and then the environment
         // Make sure the environments match, or we return 404
         if conn_header.as_bytes()[..4] != auth_header.as_bytes()[3..7] {
-            return Err(unauthorized!());
+            return Err(ApplicationError::not_found(
+                "Invalid connection header",
```
Comment (moekatib) [inline integrationos-api/src/middleware/auth.rs:28]: I think we should improve this error.

File: integrationos-api/src/middleware/auth.rs [hunk 2]
```diff
@@ -21,48 +24,41 @@ pub async fn auth(
...
-                .event_access
-                .get_one(doc! {
-                    "accessKey": key,
-                    "deleted": false
-                })
-                .await
-                .map_err(|e| InternalError::connection_error(e.as_ref(), None))?;
+    let key = auth_header
+        .to_str()
+        .map_err(|_| ApplicationError::not_found("Invalid auth header", None))?;
```
Comment (moekatib) [inline integrationos-api/src/middleware/auth.rs:36]: Same with this one. @paulkr thoughts?

File: integrationos-api/src/middleware/auth.rs (hunk 1 above)
Comment (samgj18) [reply integrationos-api/src/middleware/auth.rs:28]: I've choosen ambiguity for security

File: integrationos-api/src/middleware/auth.rs (hunk 2 above)
Comment (samgj18) [reply integrationos-api/src/middleware/auth.rs:36]: DITTO

**Review by moekatib — APPROVED** [review]

**Pull request merged**